File: app/src/main/java/com/shahu/weathery/Model/CardModel.java
package com.shahu.weathery.Model;

import android.support.annotation.NonNull;

/**
 * Created by Shahu Ronghe on 20, September, 2019
 * in Weathery
 */
public class CardModel {
    private String mName;
    private int mPosition;
    private WeatherItem mWeatherItem;
    private String mTemperature;
    private String mDescription;
    private int mCityId;

    public CardModel(String name, int position, WeatherItem weatherItem, String temperature, String description, int cityId) {
        mName = name;
        mPosition = position;
        mWeatherItem = weatherItem;
        mTemperature = temperature;
        mDescription = description;
        mCityId = cityId;
    }

    public CardModel(CardModel cardModel) {
        this.mName = cardModel.mName;
        this.mPosition = cardModel.mPosition;
        this.mWeatherItem = cardModel.mWeatherItem;
        this.mTemperature = cardModel.mTemperature;
        this.mDescription = cardModel.mDescription;
        this.mCityId = cardModel.mCityId;
    }

    public CardModel() {
    }

    public int getCityId() {
        return mCityId;
    }

    public void setCityId(int cityId) {
        mCityId = cityId;
    }

    public String getDescription() {
        return mDescription;
    }

    public void setDescription(String description) {
        mDescription = description;
    }

    public String getName() {
        return mName;
    }

    public void setName(String name) {
        mName = name;
    }

    public int getPosition() {
        return mPosition;
    }

    public void setPosition(int position) {
        mPosition = position;
    }

    public WeatherItem getImageStr() {
        return mWeatherItem;
    }

    public void setImageStr(WeatherItem imageStr) {
        mWeatherItem = imageStr;
    }

    public String getTemperature() {
        return mTemperature;
    }

    public void setTemperature(String temperature) {
        mTemperature = temperature;
    }

    @NonNull
    @Override
    public String toString() {
        return ("Position: " + this.getPosition() + ", Name: " + this.getName() + ", Temperature: " + this.getTemperature() + ", Description:" +
                " " + this.getDescription());
    }
}

File: app/src/main/java/com/shahu/weathery/model/CardModel.kt
package com.shahu.weathery.model

import android.os.Parcel
import android.os.Parcelable
import com.shahu.weathery.model.main.WeatherItem

/**
 * Created by Shahu Ronghe on 20, September, 2019
 * in Weathery
 */
class CardModel() : Parcelable {
    var name: String? = null
    var countryCode: String? = null
    var position = 0
    var weatherItem: WeatherItem? = null
    var temperature: String? = null
    var description: String? = null
    var cityId: String? = null
    var dayNight: String? = null
    var time: Long = 0
    var secondsShift = 0
    var lon: String? = null
    var lat: String? = null

    constructor(parcel: Parcel) : this() {
        name = parcel.readString()
        countryCode = parcel.readString()
        position = parcel.readInt()
        temperature = parcel.readString()
        description = parcel.readString()
        cityId = parcel.readString()
        dayNight = parcel.readString()
        time = parcel.readLong()
        secondsShift = parcel.readInt()
        lon = parcel.readString()
        lat = parcel.readString()
    }


    override fun toString(): String {
        return "Position: " + position + ", Name: " + name + ", Temperature: " + temperature + ", Description:" +
                " " + description
    }

    override fun writeToParcel(parcel: Parcel, flags: Int) {
        parcel.writeString(name)
        parcel.writeString(countryCode)
        parcel.writeInt(position)
        parcel.writeString(temperature)
        parcel.writeString(description)
        parcel.writeString(cityId)
        parcel.writeString(dayNight)
        parcel.writeLong(time)
        parcel.writeInt(secondsShift)
        parcel.writeString(lon)
        parcel.writeString(lat)
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object CREATOR : Parcelable.Creator<CardModel> {
        override fun createFromParcel(parcel: Parcel): CardModel {
            return CardModel(parcel)
        }

        override fun newArray(size: Int): Array<CardModel?> {
            return arrayOfNulls(size)
        }
    }
}
